golang/cosmos/x/vibc/keeper/keeper.go

```diff
@@ -123,7 +124,10 @@ func (k Keeper) WriteAcknowledgement(ctx sdk.Context, packet ibcexported.PacketI
 	if !ok {
 		return sdkerrors.Wrapf(channeltypes.ErrChannelCapabilityNotFound, "could not retrieve channel capability at: %s", capName)
 	}
-	return k.channelKeeper.WriteAcknowledgement(ctx, chanCap, packet, acknowledgement)
+	var ackRaw exported.Acknowledgement
+	ackRaw.Success()
+	ackRaw.Acknowledgement()
+	return k.channelKeeper.WriteAcknowledgement(ctx, chanCap, packet, ackRaw)
 }
```

Comment:
I don't see how this would work.  We need to implement the `exported.Acknowledgement` interface so that it provides exactly the raw bytes given in the `acknowledgement byte[]`.